doc/development.md

```diff
@@ -89,4 +89,205 @@
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Accepted, but better to avoid:
+- Relative name imports
+- Import many modules in a comma separated list
+- Alias an imported module with non-standard abbreviation
+
+### Indentation
+
+Use 4 spaces per indentation level.
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Correct
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+foo = long_function_name(
+        var_one, var_two, var_three, var_four)
+
+# Also with if-statements.
+if (condition_1 and
+        condition_2):
+    do_something()
+
+# One line per argument for cases where arguments expand to more than one line.
+def long_function_name(
+        var_one,
+        var_two,
+        var_three,
+        var_four):
+    print(var_one)
+
+# Hanging indents should add a level.
+foo = long_function_name(
+    var_one, var_two, var_three, var_four)
+```
+```python
+# Wrong:
+# Arguments on first line forbidden (even when vertically aligned).
+foo = long_function_name(var_one, var_two,
+                         var_three, var_four)
+
+# Further indentation required as indentation is not distinguishable.
+def long_function_name(
+    var_one, var_two, var_three, var_four):
+    print(var_one)
+
+```
+
+To break comprehension Lists, Dicts, Sets as well as generator expressions,
+keep expressions that would be written in a single line together.
+Continuation lines stick to the same rules as above. Closing
+brace/bracket/parenthesis shall line up under the first character of the
+line start.
+
+```python
+# Correct
+# Vertically aligned and extra 4 spaces for condition continuation
+foo = [var for var in bar_list
+       if condition_1 and
+           condition_2]
+
+# Closing bracket starts under the first character
+foo = [
+    var for var in bar_list if condition_1
+]
+```
+
+```python
+# Wrong
+# for-statement broken in multiple lines
+foo = [var for var
+       in bar_list if condition_1 and
+       condition_2]
+```
+
+### Docstrings
+
+Always use the three double-quote `"""` format for docstrings 
+(per [PEP 257](https://peps.python.org/pep-0257/)).
+A docstring should be organized as multiple summary lines (one physical line
+not exceeding 79 characters) terminated by a period. First and last line
+of the docstring shall be blank.
+
+Functions and classes must have a docstring, unless:
+- Not externally visible and very short.
```

Comment:
This should be relaxed to avoid having docstrings for the sake of having docstrings.  Generally, only public (non-test) entities should be required to have docstrings.  Docstrings in some non-public entities may be desirable but it depends on the qualities rather than quantities of the content and comments may be better if it is about explanation what's happening rather than how to call it.

Sometimes docstrings may not be needed even in public entities, for example, it's often impossible to provide a meaningful docstring for `def __init__(self)`.

Reply:
I mean, more or less you listed the same exceptions as I wrote:
```
- Not externally visible and very short.
- Pertains to a test module.
```

Also, here I purposely wrote about functions, not class methods. Class methods require docstring only when they are not short/obvious and externally visible. Therefore, the constructor of a class does not need a docstring, I agree. Nor do, in general, the getters/setters for class attributes, for example. What I meant for classes is that the class itself deserves a docstring:
```
class RandomClass():
    """
   This class needs a docstring.
    """
```
I will relax a bit the text and add some example.